lib/_emerge/depgraph.py

```diff
@@ -5027,6 +5027,20 @@ def _resolve(self, myfavorites):
                         if not package_is_installed:
                             continue
 
+                    # If we're emerging @selected or @world, we want to loudly warn about
+                    # no ebuilds being available for packages (bug #911180).
+                    if (
```

Comment:
There's a similar but different condition a bit below in this function but I wasn't sure if/how to merge the two.

Reply:
On reflection, I don't think that other condition can be merged with this.